### ISSUE TITLE
wcomponents-app-archetype does not seem to work

### ISSUE THREAD

**Issue by JohnMcGuinness** [BorderTech/wcomponents#247]
I built the wcomponents-app-archetype project (version 1.0.2-SNAPSHOT) and installed it locally.

I had a few issues when trying to use the archetype to create a new project, I've listed them below.

* When initially building the new project based on the archetype I received the error *'dependencies.dependency.version' for junit:junit:jar must be a valid version but is '${wcomponents.junit.version}'.*

 * **Solution:** Add the property  `<wcomponents.junit.version>4.8.2</wcomponents.junit.version>`

* I then received this compilation error: 
```
Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.5.1:compile (default-compile) on project Ache1: Compilation failure: Compilation failure:
com/github/johnmcguinness/ache1/servlet/MyAppServlet.java:[18,7] error: cannot access HttpServlet
class file for javax.servlet.http.HttpServlet not found
com/github/johnmcguinness/ache1/servlet/MyAppServlet.java:[32,4] error: method does not override or implement a method from a supertype
```

 * **Solution:** Added the following dependency: 
```
<dependency>
    <groupId>javax</groupId>
    <artifactId>javaee-web-api</artifactId>
    <version>7.0</version>
    <scope>provided</scope>
</dependency>
```

The project now successfully builds.

* I then tried to deploy to a GlassFish instance and received the following runtime exceptions when loading the app in a browser:
```
Warning:   StandardWrapperValve[myAppServlet]: Allocate exception for servlet myAppServlet
java.lang.ClassNotFoundException: org.apache.commons.logging.LogFactory
``` 
and 
```
Warning:   StandardWrapperValve[myAppServlet]: Allocate exception for servlet myAppServlet
java.lang.ClassNotFoundException: org.apache.commons.configuration.Configuration
```

 * **Solution:** Added the following dependencies: 
```
<dependency>
    <groupId>commons-logging</groupId>
    <artifactId>commons-logging</artifactId>
    <version>1.1.1</version>
</dependency>
``` 
and

```
<dependency>
    <groupId>commons-configuration</groupId>
    <artifactId>commons-configuration</artifactId>
    <version>1.6</version>
</dependency>
```

* I then received this runtime exception: `Info:   Service called on com.github.johnmcguinness.ache1.servlet.MyAppServlet@409b5746... from /Ache1/app
Severe:   Caught exception during action phase.
java.lang.NoClassDefFoundError: org/apache/commons/fileupload/FileUploadException
	at com.github.bordertech.wcomponents.servlet.HttpServletHelper.createRequest(HttpServletHelper.java:228)`

 * **Solution:** Added this dependency: `<dependency>
			<groupId>commons-fileupload</groupId>
			<artifactId>commons-fileupload</artifactId>
			<version>1.3.1</version>
		</dependency>`

* The app now runs without throwing exceptions, but does not render anything. It includes these warnings in the log: `Warning:   Could not load theme version properties file "/theme/default/version.properties"
Warning:   build.number property not found in "/theme/default/version.properties". Will use wcomponent project version.`

 * **Solution:** Added this dependency 
```
<dependency>
    <groupId>com.github.bordertech.wcomponents</groupId>
    <artifactId>wcomponents-theme</artifactId>
    <version>${bordertech.wcomponents.version}</version>
</dependency>
```

* Still not rendering. Checked in the theme jar and realised the theme is not on the path as displayed in the warning.

 * **Solution:** In wcomponents-app.properties updated the value of `bordertech.wcomponents.theme.name` from `default` to `wcomponents-theme`.

The app now runs successfully.

**Comment by marksreeves** [comment]
This is a duplicate of #13

**Comment by marksreeves** [comment]
@JohnMcGuinness It appears you addressed this in #248. If this pull fixes this bug can you please close this issue.

**Comment by JohnMcGuinness** [comment]
Fixes #248 @marksreeves

**Comment by JohnMcGuinness** [comment]
Theme not working properly.

**Comment by marksreeves** [comment]
Fixed by #276
